vignettes/hypothesis_testing.Rmd

```diff
@@ -0,0 +1,129 @@
+1. Each of the remaining submatrices of L associated with an effect $F_2$ that contains $F_1$ is $(X_1^\top M X_1)^{-}X_1^\top M X_2$
+
+where $X_0$ stands for columns of $X$ whose associated effect do not contain $F_1$,
+$X_1$ stands for columns of $X$ associated with $F_1$,
+$X_2$ stands for columns of $X$ whose associated effect contains $F_1$,
+$M = I - X_0(X_0^\top X_0)^{-}X_0^\top$,
+and $Z^{-}$ stands for the g2 inverse of $Z$.
+
+Note: In `mmrm` package, we do not allow singularity in general, so the g2 inverse is the inverse.
+Thus we can use identity matrix in step 2.
+
+# Type III Hypothesis Testing
+
+For type III hypothesis testing, we assume that the hypothesis tested should be the same for all
+designs with the same general form of estimable functions.
+It can be defined with the following step:
+
+Create a contrast matrix $L$, with rows equal to the number of parameters associated with the effect,
+columns equal to the number of all parameters.
+
+1. All columns of $L$ associated with effect not containing $F_1$ (except $F_1$) are set to 0.
+1. The submatrix of $L$ associated with $F_1$ is set to identity matrix.
+1. For each of the remaining submatrices of L associated with an effect $F_2$ that contains $F_1$, equate the
+values.
+
+Here equate the values means that, given that the coefficients in step 2 are all 1 (diagonal), in step 3
+this 1 is divided by the total number of new levels.
+
+# Hypothesis Testing in SAS
+
+In `PROC MIXED` we can specify `HTYPE=1,2,3` to enable these hypothesis testings, and we can use option `e1 e2 e3` to
+view the estimable functions.
+
+## Special Considerations
+
+For `PROC MIXED`, it is important that the categorical values have the correct order, especially
+for structured covariance (otherwise the correlation between visits can be messed up).
+We usually use `CLASS variable(REF=)` to define the reference level.
+However, SAS will put the reference level in the last, so if the visit variable is included in fixed effect,
+there can be issues with the comparison between SAS and R, as in R we still use the first level as our reference.
+
+When we conduct the hypothesis testing, we will also face this issue.
+The result will be slightly different because of different reference groups, however, if correct contrast is used
+we should have closer result.
+
+### Why Model Covariate Order Changes My Testing in SAS?
+
+For type II/III testing, it is true that the order of covariate should not affect the result.
+However, if there is an interaction term, the reference level can be changed.
+With different reference level the result will be slightly different.
+
+### Why `mmrm` Gives More Covariates Than SAS?
+
+For `PROC MIXED`, if a model is specified as follows
+
+```
+MODEL FEV1 = ARMCD SEX ARMCD * SEX ARMCD * FEV1_BL
+```
+
+Then it is equivalent to the following `mmrm` model
+
+```
+FEV1 ~ ARMCD * SEX + ARMCD * FEV1_BL - FEV1_BL
```

Comment:
```suggestion
FEV1 ~ ARMCD * SEX + ARMCD:FEV1_BL
```
would be simpler

Reply:
this is on purpose to use `ARMCD*FEV1_BL` to better illustrate the difference: in SAS using `ARMCD*FEV1_BL` will only add the interaction term, but not the two main effect.

Reply:
can we then add a comment that in R you can do the "ARMCD:FEV1_BL" instead?

Reply:
added